### PR TITLE
feat(auto-build): option to not publish build logs when build is successful

### PR DESCRIPTION
**Pull request by jogold**
Removes noise in PR comments.

I bet those logs are hardly ever checked.

See https://github.com/jlhood/github-codebuild-logs/pull/29

-----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: lib/auto-build.ts [hunk 1]
```diff
@@ -29,6 +29,15 @@ export interface AutoBuildOptions {
...
   readonly publicLogs?: boolean;
 
+  /**
+   * Whether to publish a link to build logs when build is successful.
+   *
+   * @see https://github.com/jlhood/github-codebuild-logs#app-parameters
+   *
+   * @default false
+   */
+  readonly publicLogsOnSuccess?: boolean;
```
Comment (eladb) [inline lib/auto-build.ts:39]: Let's make the default true. I find it very useful to get an email when my build passes.

**Review by eladb — CHANGES_REQUESTED** [review]

File: lib/auto-build.ts (hunk 1 above)
Comment (jogold) [reply lib/auto-build.ts:39]: 😄 this is exactly the kind of noise I was referring to in the PR description 😄 

I personally don't like to get 5 emails when I push 5 different commits to a PR... and I like to only get alerts/notifications when something fails (this is the behavior of failed GitHub workflows).

Tell me if you still want to change this default, I will of course adapt accordingly 😉

File: lib/auto-build.ts [hunk 2]
```diff
@@ -90,11 +99,12 @@ export class AutoBuild extends Construct {
       new serverless.CfnApplication(this, 'GitHubCodeBuildLogsSAR', {
         location: {
           applicationId: 'arn:aws:serverlessrepo:us-east-1:277187709615:applications/github-codebuild-logs',
-          semanticVersion: '1.3.0',
+          semanticVersion: '1.4.0',
```
Comment (NetaNir) [inline lib/auto-build.ts:102]: Oh we really need this upgrade... (we really need this to be configurable, and we need code build to do be better but I'll take what I can get)

File: lib/auto-build.ts (hunk 1 above)
Comment (eladb) [reply lib/auto-build.ts:39]: Let's change the default to true

**Review by eladb — APPROVED** [review]

**Pull request merged**